### PR TITLE
KEP proposals for adding Alibaba Cloud Provider into kubernetes ecosystem

### PR DESCRIPTION
**Pull request by aoxn**
Alibaba Cloud provides the Cloud Provider interface implementation as an out-of-tree cloud-controller-manager. It allows Kubernetes clusters to leverage the infrastructure services of Alibaba Cloud .
It is original open sourced project is [https://github.com/AliyunContainerService/alicloud-controller-manager](https://github.com/AliyunContainerService/alicloud-controller-manager)

Here we propose a repository under kubernetes orgnization to host Alibaba Cloud Provider code . It would be great benefit for people who want to run kubernetes on Alibaba Cloud.

### CODE REVIEW

File: keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+### Repository Requirements
+
+Alibaba Cloud Providers is a working implementation of the [Kubernetes Cloud Controller Manager](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/).
+
+The repo requirements is mainly a copy from [cloudprovider KEP](https://github.com/kubernetes/community/blob/master/keps/sig-cloud-provider/0002-cloud-controller-manager.md#repository-requirements). Open the link for more detail.
+
+### User Experience Reports
+
+User experience reports is highly encouraged. User can submit any experience issue through [Aliyun Container Service](https://github.com/AliyunContainerService/alicloud-controller-manager/issues) github pages.
```
Comment (andrewsykim) [inline keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md]: Can you reword this to: 

```
Usage of aliyun container services can be seen from github issues in the existing alicloud controller manager repo: https://github.com/AliyunContainerService/alicloud-controller-manager/issues
```

**Comment by hogepodge** [comment]
My primary question regarding the KEP is, do we need you to state your goals for testing and documentation, or does the sig-cloud-provider umbrella imply that? I'm happy to see the getting started doc in the repository, but testing.md is empty.

**Comment by andrewsykim** [comment]
IMO I think we can set that as an official requirement once we have more cloud providers setting a good example for what this looks like. I know OpenStack has been doing some good work there but I'd personally like to see more (in-tree) providers showcasing this before we make it a hard requirement for new providers.

**Comment by aoxn** [comment]
I personally think that It is a good practice for cloudprovider to follow the unified integration test procedure. So as Alibaba Cloud Provider.  Also , I can update the testing.md doc with some unit-test. Thanks

File: keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+## Prerequisites
+
+1. The VPC network is supported in this project. The support for classic network or none ECS environment will be out-of-scope. 
+2. When using the instance profile for authentication, an instance role is required to attach to the ECS instance firstly.
+3. Kubernetes version v1.7 or higher
+
+### Repository Requirements
+
+Alibaba Cloud Providers is a working implementation of the [Kubernetes Cloud Controller Manager](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/).
```
Comment (andrewsykim) [inline keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md]: `Alibaba Cloud Provider` -> `[Alibaba Cloud Controller Manager](https://github.com/AliyunContainerService/alicloud-controller-manager)`

File: keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+The repo requirements is mainly a copy from [cloudprovider KEP](https://github.com/kubernetes/community/blob/master/keps/sig-cloud-provider/0002-cloud-controller-manager.md#repository-requirements). Open the link for more detail.
+
+### User Experience Reports
+
+Usage of aliyun container services can be seen from github issues in the existing alicloud controller manager repo: https://github.com/AliyunContainerService/alicloud-controller-manager/issues
```
Comment (andrewsykim) [inline keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md]: It may not hurt to mention Alibaba's CNCF membership, that's usually also a pretty good indication of user adoption and support in the community.

**Comment by andrewsykim** [comment]
Left a few minor comments, LGTM otherwise

**Comment by aoxn** [comment]
@andrewsykim  @hogepodge  Refine doc and Add testing.md.

File: keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  - "@andrewsykim"
+  - "@hogepodge"
+  - "@jagosan"
+editor: TBD
+creation-date: 2018-06-20
+last-updated: 2018-06-20
+status: provisional
+see-also:
+  - KEP-2
+  - KEP-4
```
Comment (andrewsykim) [inline keps/sig-cloud-provider/providers/0005-cloud-provider-alibaba-cloud.md]: I'll fix these in a follow up PR

**Comment by andrewsykim** [comment]
@aoxn sorry one more thing, can you squash your commits please?

**Comment by aoxn** [comment]
@andrewsykim  Done

**Comment by andrewsykim** [comment]
FYI this KEP is still in the `provisional` status, see https://github.com/kubernetes/community/pull/2449
